test/plugin.test.js

```diff
@@ -1028,3 +1028,27 @@ test('plugin metadata - version not matching requirement 3', t => {
     done()
   }
 })
+
+test('plugin metadata - release candidate', t => {
+  t.plan(2)
+  const fastify = Fastify()
+  Object.defineProperty(fastify, 'version', {
+    value: '99.0.0-rc.1'
+  })
+
+  plugin[Symbol.for('plugin-meta')] = {
+    name: 'plugin',
+    fastify: '99.x'
+  }
```

Comment:
I think this should specify `fastify: '^98.0.0'`.

Reply:
No this is expected and the current case.

Reply:
I still don't understand, can you open a fresh issue if you think it's a bug?